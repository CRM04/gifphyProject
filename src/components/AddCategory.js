import React, { useState } from 'react';
import propTypes from 'prop-types';


export const AddCategory = ({ setHeroes }) => {
    const [inputValue, setinputValue] = useState('');

    const inputValueChange = (e) => {
        setinputValue(e.target.value);
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        if (inputValue.trim().length > 1) {
            setHeroes(her => [inputValue, ...her]);
            setinputValue('');
        }
    }

    return (
        <form onSubmit={handleSubmit}>
            <p> {inputValue} </p>
            <input type="text" value={inputValue} onChange={inputValueChange} ></input>
        </form>
    );

}

AddCategory.propTypes = {
    setHeroes: propTypes.func.isRequired
}

