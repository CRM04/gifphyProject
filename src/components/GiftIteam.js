import React from 'react';
import propTypes from 'prop-types';

export const GiftItem = ({ title, url }) => {

    return (
        <div className="card">
            <img src={url} alt={title} ></img>
            <p> {title} </p>
        </div>
    );
}

GiftItem.prototypes = {
    title: propTypes.string.isRequired,
    url: propTypes.string.isRequired
}