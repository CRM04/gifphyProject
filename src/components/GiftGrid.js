import React from 'react';
import { useFetchGif } from '../hooks/useFetchGift';
import { GiftItem } from './GiftIteam';

export const GiftGrid = ({ heroe }) => {

    const { data, loading } = useFetchGif(heroe);

    return (
        <>
            <h3 className="animate__animated animate__fadeIn"> {heroe} </h3>
            { loading && <p>Loading... </p> }
            <div className="card-grid animate__animated animate__fadeIn">
                {
                    data.map(gif => <GiftItem {...gif} key={gif.id} />)
                }
            </div>
        </>
    );
}