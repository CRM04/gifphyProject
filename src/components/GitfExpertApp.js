import React, { useState } from 'react';
import { AddCategory } from './AddCategory';
import { GiftGrid } from './GiftGrid';

const GiftExpertApp = ( {categories = []} ) => {

    const [Heroes, setHeroes] = useState(categories);

    return (
        <div>
            <h2> GitfExpertApp </h2>
            <AddCategory setHeroes={setHeroes} />
            <hr></hr>
            {Heroes.map(heroe => <GiftGrid heroe={heroe} key={heroe} />)}
        </div>
    );


}

export default GiftExpertApp;