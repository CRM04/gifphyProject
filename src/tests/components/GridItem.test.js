import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import { GiftItem } from '../../components/GiftIteam';

describe('GridItem test suit', () => {
    const title = "A title";
    const url = 'http://localhost/image.jpg';

    let wrapper = shallow(<GiftItem title={title} url={url} />);

    test('GridItem snapshot should match', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('GridItem paragraph should contain the title', () => {
        const text = wrapper.find('p').text();
        expect(text.trim()).toBe(title);
    });

    test('img should have src and alt', () => {
        const { src, alt } = wrapper.find('img').props();
        expect(src).toBe(url);
        expect(alt).toBe(title);
    });

    test('main div should have the right class', () => {
        const { className } = wrapper.find('div').props();
        expect( className ).toBe('card');
    });

});


