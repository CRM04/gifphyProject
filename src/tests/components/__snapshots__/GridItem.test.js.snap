// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridItem test suit GridItem snapshot should match 1`] = `
<div
  className="card"
>
  <img
    alt="A title"
    src="http://localhost/image.jpg"
  />
  <p>
     
    A title
     
  </p>
</div>
`;
